lib/autoupdate/agent/updater.go

```diff
@@ -1053,10 +1062,23 @@ func (u *Updater) update(ctx context.Context, cfg *UpdateConfig, target Revision
 // Setup writes updater configuration and verifies the Teleport installation.
 // If restart is true, Setup also restarts Teleport.
 // Setup is safe to run concurrently with other Updater commands.
-func (u *Updater) Setup(ctx context.Context, path string, restart bool) error {
-	// Setup teleport-updater configuration and sync systemd.
+func (u *Updater) Setup(ctx context.Context, path string, installSELinux, restart bool) error {
+	// The SELinux module needs to be removed if it was previously
+	// installed but is not supposed to be installed now.
+	var removeSELinux bool
+	if !installSELinux {
+		cfg, err := readConfig(u.UpdateConfigFile)
```

Comment:
Let's avoid reading this file from `setup`. This circumvents update.lock.

Instead of trying to figure out which state change is needed, can we always attempt to remove when installSELinux=false? Then we can also avoid passing two bools to SetupNamespace

Reply:
The problem with unconditionally trying to remove the module is when `teleport-update` is run on Debian or a Linux distro we don't support, the logs will be polluted with SELinux module removal errors that don't make sense. Or if a user is running on RHEL but doesn't want to have our module installed or doesn't have the necessary packages installed to manage modules, `teleport-update` will always try and fail to remove the module.

Reply:
Can you detect if the module is installed before trying to remove it?

Reply:
Yes, but AFAIK it requires `semodule` which runs into the same issues. Knowing Linux there's got to be a file somewhere that has a list of installed modules, but idk if it's easily readable.

Reply:
Let's find a way to accomplish this without relying on the old configuration. There are a number of cases where this won't work, e.g., if installation failed after the module was successfully installed.

Reply:
I found a way to easily see if the module is installed, removed read to config here